crates/core/src/host.rs

```diff
@@ -43,9 +44,9 @@ pub struct HostData {
     pub link_definitions: Vec<InterfaceLinkDefinition>,
     /// list of cluster issuers
     pub cluster_issuers: Vec<ClusterIssuerKey>,
-    /// Merged named configuration set for this provider at runtime
+    /// Merged named configuration set for this provider at runtime, indexed by link name
     #[serde(default)]
-    pub config: HashMap<String, String>,
+    pub configs: HashMap<LinkName, HashMap<String, String>>,
```

Comment:
Re: above comment, I think the config should stay the same, but this hashmap of link name -> config should replace `link_definitions`

Reply:
OK I've reverted this for now -- just the `config` that comes in gets used